[Zhang Wanlin] iP

WadeForward opened the pull request.

Commit:
Add Gradle support

WadeForward committed:
Level-1. Greet, Echo, Exit

WadeForward committed:
Level-1 version 2

WadeForward committed:
Level-2 version 1

WadeForward committed:
Level-2 version 2:
fixed some incomplete statements.

WadeForward committed:
Level-3 Version 1

WadeForward committed:
Level-2 Version 3: exception when list is empty fixed

WadeForward committed:
Merge branch 'master' of https://github.com/ZhangWanlin98/ip

# Conflicts:
#	src/Duke.java
#	src/Memory.java

WadeForward committed:
Level-3 Version 2

WadeForward committed:
Level-4 version 1

WadeForward committed:
Merge branch 'master' of https://github.com/ZhangWanlin98/ip

# Conflicts:
#	src/Duke.java
#	src/Memory.java
#	src/Status.java

WadeForward committed:
Level 4 Version 2

WadeForward committed:
A-TextUiTesting version 1

WadeForward committed:
A-TextUiTesting Version 2

WadeForward committed:
Level-5 version 1

WadeForward committed:
Level-6 version 1

WadeForward committed:
A-TextUiTesting version

WadeForward committed:
Level-4 version 3 or Level 5 version

WadeForward committed:
Level-6 version 2

WadeForward committed:
Update Level-5

Level-5 is featured because it has more error handling compard to previous levels.

The Level-5 debugging return has an improper return when the deadline and event omit time.

This mistake has been fixed with change in Operation class.

WadeForward committed:
Update Level-6

Level-6 has new feature of delete function added into the command of Duke.

Simlar to Level-5 update, the improper return when deadline or event omits the time element, is carried foward to last update of Level-6.

The mistake output has been fixed by changing the Operation class.

WadeForward committed:
Update Level-7

At Level-6, Duke has all the basic functions of a reminder.

However, Duke cannot store all information into the hardware, which makes it meaningless if it functions as a reminder.

Update is made to include the use of hardware. Now, with the use of adding event, list, done, delete, corresponding information in ./data/Duke.txt will also change accordingly.

WadeForward committed:
Update Level-7

Current stage has fulfiled all required tasks.

The EditiFile class should be named as EditFile.

This typo has been fixed.

WadeForward committed:
Update Level-8

For Level-7 and levels before, event and deadline tasks need to specify the respective time.

However, this time can be in any words, which sometimes does not even make sense.

In this level, the format of time has been formated and the user must follow the time format in order to note down the task.

WadeForward committed:
Merge branch 'branch-Level-7' into master

* branch-Level-7:
  Update Level-7
  Update Level-7

WadeForward committed:
Merge branch 'branch-Level-8' into master

* branch-Level-8:
  Update Level-8

# Conflicts:
#	data/Duke.txt
#	src/DukeException.java
#	src/Exceptions.java
#	src/Formating.java

WadeForward committed:
Update A-MoreOOP

Currently, there are a number of classes that have different functions to serve the clients.

However, some classes are too powerful and do not serve single purpose.

A more OOP desgin is applied in the code to make each class to serve a single purpose. (i.e. the required Storage class is divided into WriteIn.java, ReadFile,java, EditFile.java to serve their function more explicitly).

WadeForward committed:
Update A-packages

All classes are just under src, and only the file name tells the function of this java file.

This makes it not clear enough about the purpose of each class.

This commit shifted all classes into different packages to clarify further about the collective function of classes.

WadeForward committed:
Update A-JUnit

All previous levels are fulfilled.

However, there is no test method.

JUnit test files are added into the ip/src/test.

WadeForward committed:
Update A-JavaDoc

All tested code function as expected.

No comments is here to clarify the functino of each method and class.

comments are added.

WadeForward committed:
Update A-CodingStandard

The code is functional.

Yet, the code is not standardized.

The commit standardized the code.

WadeForward committed:
Update Level-9

The code is functional.

But there is no searching function.

Hence, this commit added this feature.

WadeForward committed:
Merge branch 'branch-A-CodingStandard' into master

# Conflicts:
#	src/Duke/main/Parser.java

Source used: Update Level-9

WadeForward committed:
Merge branch 'branch-A-JavaDoc' into master

* branch-A-JavaDoc:
  Update A-JavaDoc

# Conflicts:
#	src/Duke/exception/DukeException.java
#	src/Duke/main/Parser.java
#	src/Duke/storage/WriteIn.java

WadeForward committed:
Update Level-9

The current level fulfils the functionality requirements.

However, comments are missing.

Hence, this commit adds on the comments.

WadeForward committed:
Merge branch 'branch-Level-9' into master

* branch-Level-9:
  Update Level-9

# Conflicts:
#	src/Duke/exception/DukeException.java
#	src/Duke/main/Format.java
#	src/Duke/main/Parser.java
#	src/Duke/main/Response.java
#	src/Duke/main/Statement.java
#	src/Duke/main/Time.java
#	src/Duke/main/Ui.java
#	src/Duke/storage/EditFile.java
#	src/Duke/storage/ReadFile.java
#	src/Duke/storage/WriteIn.java
#	src/Duke/task/Deadline.java
#	src/Duke/task/Event.java
#	src/Duke/task/Task.java
#	src/Duke/task/TaskList.java
#	src/Duke/task/Todo.java
#	src/test/java/FormatTest.java

GabriellaTeh commented on src/Duke/exception/DukeException.java:
```diff
@@ -1,73 +1,45 @@
-        System.out.println(
-                new Formating<>(Exceptions.TIMEFORMATEXCEPTION.toString())
-        );
+        System.out.println(new Format<>(Exceptions.TIMEFORMATEXCEPTION.toString()));
```
The purpose of the method would be clearer with javadoc comments for each method.

WadeForward replied:
Thank you very much for pointing out my errors!!!
I have tried my best to make improvement and made it as my latest commit.

GabriellaTeh commented on src/Duke/main/Format.java:
```diff
@@ -114,22 +114,23 @@ public Task stringToTask() {
+                    Statement.TASKADDED.toString() +
+                    content +
+                    FormatString.NEXTLINE.toString() +
+                    String.format(Statement.REPORT.toString(), Parser.taskList.getTaskList().size()) +
```
This line should be indented with 8 spaces.

GabriellaTeh commented on src/Duke/main/Parser.java:
```diff
@@ -99,25 +95,18 @@ public static void done(String order) {
 
                 System.out.println(
-                        new Formating<>(new Response(Status.DONE.toString() + task)));
+                        new Format<>(new Response(Statement.DONE.toString() + task)));
```
"task" should be indented with 8 spaces.

GabriellaTeh commented on src/Duke/main/Parser.java:
```diff
@@ -130,44 +119,50 @@ public static void delete(String order) {
                                 task + "\n" +
                                 String.format
-                                        (Status.REPORT.toString(), taskList.getTaskList().size());
+                                        (Statement.REPORT.toString(), taskList.getTaskList().size());
```
This should be indented with 8 spaces.

WadeForward committed:
Update Level-9

Current version has all functionality.

Only delete command can remove the tasks one by one. It may be slow.

clear command is added to clear the tasks in one go.

GabriellaTeh commented on src/Duke/storage/EditFile.java:
```diff
@@ -67,31 +72,31 @@ public void deleteLine(int lineNum) {
             File fileToBeModified = new File(this.path);
-            String text = "";
+            StringBuilder text = new StringBuilder();
             BufferedReader reader = new BufferedReader(new FileReader(fileToBeModified));
```
This line should be indented with 8 spaces.

GabriellaTeh commented on src/test/java/FormatTest.java:
```diff
@@ -11,20 +11,21 @@
-                new Formating<>("   hahaha  ")
+                new Format<>("   hahaha  ")
                 .shorten()
                 .getContent();
```
These two lines should be indented with 4 more spaces.

GabriellaTeh reviewed (commented):
Overall looks good except some indentation errors. :-)

WadeForward committed:
Merge remote-tracking branch 'origin/add-gradle-support' into A-Gradle

WadeForward committed:
Update A-Gradle

Gradle is used to automate project.

WadeForward committed:
Update A-CheckStyle

Gradle is added into the project.

The coding style is not checked yet.

This commit checked and corrected the coding style.

WadeForward committed:
Update A-CheckStyle

The checkStyle is cleared.

There is an hidden coding style error.

The commit corrected it.

WadeForward committed:
Update Level-10

Based on the codes in the last commit,

GUI is added.

Sruthisarav commented on src/main/java/duke/exception/Exceptions.java:
```diff
@@ -69,5 +69,13 @@ public String toString() {
+        @Override
+        public String toString() {
+            return "    OOPS!!! So what would you like to find?\n"
+                    + "    Try to add something behind 'find'.";
```
You could combine lines 77 and 78. There is no need to separate them

WadeForward replied:
Thank you very much for pointing out my errors!!!
I have tried my best to make improvement and made it as my latest commit.

Sruthisarav commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,64 @@
+    public String greet() {
+        return new Format<>(
+                new Response(Statement.GREET.toString()
+                )
```
Lines 48 to 50 could have been in a single line.

Sruthisarav commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,64 @@
+    public String exit() {
+        return new Format<>(
+                new Response(Statement.BYE.toString()
+                )
```
Lines 59 to 61 could have been in the same line too

Sruthisarav reviewed (commented):
Overall, good work! I don't see any coding standard violations except for those in brackets. You need at least one parameter inside the bracket before separating the rest with new lines.
Example: 
```
method_name(param1,

param2,

param3)
```

You tend to do this instead:
```
method_name(
param1,

param2,

param3
)
```

WadeForward committed:
Update Level-10

Previous commit contains some indentation errors and lack of java doc for some methods.

This commit fixed it.

WadeForward committed:
Update Level-10

Pending to make more adjustments.

WadeForward committed:
Update Level-10

Current state stops at Gradle fat jar file.

WadeForward committed:
Update Level-10

Although the function works fine in IDE

When converted to jar, storage class has problem during operation.

All related bugs are fixed and GUI is improved.

WadeForward committed:
Update A-Assertions

Assertions are added to ensure the correct logic in the operation of program.

WadeForward committed:
Merge pull request #1 from ZhangWanlin98/A-Assertions

Update A-Assertions

WadeForward committed:
Increment A-CodeQuality

This commit checked the code quality and to correct the incorrect parts.

WadeForward committed:
Merge pull request #2 from ZhangWanlin98/A-CodeQuality

Increment A-CodeQuality

WadeForward committed:
Increment A-CI

This is more on a test nature in which it is unsure if the .yml is functional.

WadeForward committed:
Update A-CI

The code has been implememted into gradle.yml file.

WadeForward committed:
Add update in the for of update <task index> <time/detail> /to <detailToUpdate>.

WadeForward committed:
Update A-CI

file .workflows is renamed as workflow and shifted to be under .github.

WadeForward committed:
Merge commit '9d6838d81fb4f37e896da6eae572f00b8c6e7ad2' into master

* commit '9d6838d81fb4f37e896da6eae572f00b8c6e7ad2':
  Add update in the for of update <task index> <time/detail> /to <detailToUpdate>.

WadeForward committed:
Add a better GUI

Features like different chatBox colour, auto close are added.

WadeForward committed:
Update better GUI and better OOP

Framworks and alignments are added

WadeForward committed:
Update BetterGui and code quality.

Some checkStyple is missed.

This commit has fixed it.

WadeForward committed:
add user guide

WadeForward committed:
Update UG

more examples of update function are shown

WadeForward committed:
Update Format class

counter of tasks is not updated

this commit fixed it

WadeForward committed:
update A-CI

WadeForward committed:
update UpdateCommand class

removed unused import class

WadeForward committed:
update test cases for parser and format , and workflows

bugs were found in test cases.

This commit fixed them

WadeForward committed:
update DeleteCommand, readme and workflow

Previously the delete command cannot review correct task number.
Hence readme is also corrected.

WadeForward committed:
Update ClearCommand and UG

clear all is added to clear completed tasks

WadeForward committed:
update UG bookmark

WadeForward committed:
add UG theme

WadeForward committed:
delete manual theme for UG

WadeForward committed:
Set theme jekyll-theme-architect

WadeForward committed:
Update Time class

Features of human time description, like today morning, tomorrow night are added.

WadeForward committed:
Merge branch 'master' of https://github.com/ZhangWanlin98/ip into master

WadeForward committed:
update Time class

deleted some unused code.

WadeForward committed:
update README
